test/li_test.jl

```diff
@@ -0,0 +1,144 @@
+using SpecialFunctions
+using Base.Test
+
+const SF = SpecialFunctions
+
+
+# golden ratio is used in some tests
+ϕ = (sqrt(5)+1)/2.0
+
+@testset "Polylogarithm Li" begin
+    @testset "Check different input formats" begin
+        @test SF.Li(-1.0, 0.0) ≈ 0.0
+        @test SF.Li(-1, 0.0) ≈ 0.0
+        @test SF.Li(-1.0, Complex(0.0)) ≈ 0.0 + 0.0im # don't seem to need imaginary bit here
+        @test SF.Li(Complex(-1.0), 0.0) ≈ 0.0 + 0.0im
+        @test SF.Li(Complex(-1.0), 0.0) ≈ 0.0
+        @test SF.Li(1, 0) ≈ 0.0
+        x = collect(0.0:0.1:0.9)
+        @test all([SF.Li.(1, x)[i] ≈ SF.Li(1, x[i]) for i=1:length(x)])
+        @test SF.Li(Complex(-1.0), Complex(0.3)) ≈ SF.Li(-1.0, 0.3)
```

Comment:
I saw you used  ≅ but I didn't know the distinction, so thanks!

I will add additional checks. 

I did pretty much exactly as suggested -- I had some code checking against mpmath. There are errors, and the grow as Im(s) increases.

However, despite its claims I lost a little faith in mpmaths accuracy for problems in this domain when it failed a couple of simple Bernoulli polynomial tests (e.g., one of the zeros wasn't)

mpmath is the only easy way I had to mass produce values. I will pull a few off alpha, but how many is enough?